modP2pImpl/src/org/aion/p2p/impl1/P2pMgr.java

```diff
@@ -992,79 +154,75 @@ public void run() {
-                Thread t = new Thread(new TaskReceive(), "p2p-worker-" + i);
+                Thread t = new Thread(getReceiveInstance(), "p2p-worker-" + i);
                 t.setPriority(Thread.NORM_PRIORITY);
                 t.start();
             }
 
             if (upnpEnable)
-                scheduledWorkers.scheduleWithFixedDelay(new TaskUPnPManager(selfPort), 1, PERIOD_UPNP_PORT_MAPPING,
+                scheduledWorkers.scheduleWithFixedDelay(
+                        new TaskUPnPManager(selfPort),
+                        1,
+                        PERIOD_UPNP_PORT_MAPPING,
                         TimeUnit.MILLISECONDS);
 
             if (showStatus)
-                scheduledWorkers.scheduleWithFixedDelay(new TaskStatus(), 2, PERIOD_SHOW_STATUS, TimeUnit.MILLISECONDS);
+                scheduledWorkers.scheduleWithFixedDelay(
+                    getStatusInstance(),
+                    2,
+                    PERIOD_SHOW_STATUS,
+                    TimeUnit.MILLISECONDS);
 
             if (!syncSeedsOnly)
-                scheduledWorkers.scheduleWithFixedDelay(new TaskRequestActiveNodes(this), 5000,
-                        PERIOD_REQUEST_ACTIVE_NODES, TimeUnit.MILLISECONDS);
+                scheduledWorkers.scheduleWithFixedDelay(
+                        new TaskRequestActiveNodes(this),
+                        5000,
+                        PERIOD_REQUEST_ACTIVE_NODES,
+                        TimeUnit.MILLISECONDS);
 
-            Thread thrdClear = new Thread(new TaskClear(), "p2p-clear");
+            Thread thrdClear = new Thread(getClearInstance(), "p2p-clear");
             thrdClear.setPriority(Thread.NORM_PRIORITY);
             thrdClear.start();
 
-            Thread thrdConn = new Thread(new TaskConnectPeers(), "p2p-conn");
+            Thread thrdConn = new Thread(getConnectPeersInstance(), "p2p-conn");
             thrdConn.setPriority(Thread.NORM_PRIORITY);
             thrdConn.start();
 
         } catch (IOException e) {
-            if (showLog)
-                System.out.println("<p2p tcp-server-io-exception>");
+            if (showLog) System.out.println("<p2p tcp-server-io-exception>");
         }
     }
 
-    @Override
-    public INode getRandom() {
-        return nodeMgr.getRandom();
-    }
-
-    @Override
-    public Map<Integer, INode> getActiveNodes() {
-        return new HashMap<>(this.nodeMgr.getActiveNodesMap());
-    }
-
-    public int getTempNodesCount() {
-        return nodeMgr.tempNodesSize();
-    }
-
     @Override
     public void register(final List<Handler> _cbs) {
         for (Handler _cb : _cbs) {
             Header h = _cb.getHeader();
             short ver = h.getVer();
             byte ctrl = h.getCtrl();
             if (Ver.filter(ver) != Ver.UNKNOWN && Ctrl.filter(ctrl) != Ctrl.UNKNOWN) {
-                if (!versions.contains(ver)) {
-                    versions.add(ver);
-                }
+                versions.add(ver);
```

Comment:
Is this change intended?

Reply:
since versions is a set the add call won't allow duplicates